fe/fe-core/src/main/java/org/apache/doris/statistics/Statistics.java

```diff
@@ -159,6 +164,26 @@ public void updateColumnStats(long tableId, String partitionName, String columnN
         }
     }
 
+    public void dropTableStats(long tableId) {
+        dropPartitionStats(tableId, null);
+    }
+
+    public void dropPartitionStats(long tableId, String partitionName) {
```

Comment:
you can simply add synchronized modifier to this method, since whole code in it is in the synced block

Reply:
Besides, I think the ` if (idToTableStats.containsKey(tableId))` is a bit redundant, since you will use the value anyway in the inner block